lib/bloc/account/add_funds_bloc.dart

```diff
@@ -11,19 +14,65 @@ class AddFundsBloc {
   final _addFundResponseController = new StreamController<AddFundResponse>();
   Stream<AddFundResponse> get addFundResponseStream => _addFundResponseController.stream;
 
+  final _moonPayOrderController = new StreamController<MoonpayOrder>();
+  Sink<MoonpayOrder> get moonPayOrderSink => _moonPayOrderController.sink;
+  Stream<MoonpayOrder> get moonPayOrderStream => _moonPayOrderController.stream;
+
+  final _orderController = new BehaviorSubject<MoonpayOrder>();
+  Sink<MoonpayOrder> get orderSink => _orderController.sink;
+
   AddFundsBloc(String userID){
-    BreezBridge breezLib = new ServiceInjector().breezBridge;
+    ServiceInjector injector = ServiceInjector();
+    BreezBridge breezLib = injector.breezBridge;
     _addFundRequestController.stream.listen((request){
       _addFundResponseController.add(null);
         breezLib.addFundsInit(userID)
           .then((reply) => _addFundResponseController.add(new AddFundResponse(reply)))
           .catchError(_addFundResponseController.addError);
       })
       .onDone(_dispose);
+
+    _initializeWithMoonpayOrder(injector);
+    _listenToMoonpayOrder(injector);
+  }
+
+  _initializeWithMoonpayOrder(ServiceInjector injector) {
```

Comment:
I would simplify this function by always adding the last `MoonpayOrder` to the stream and no need to delete it from the storage as well.
Something like:
```suggestion
  var order = preferences.getString("pendingMoonpayOrderAddress");
  if (order != null) {
      _moonPayOrderController.add(MoonpayOrder(
            preferences.getString("pendingMoonpayOrderAddress") ?? null, 
            preferences.getInt("pendingMoonpayOrderTimestamp") ?? null))
   }
```

The UI only needs to filter those expired (already does that).

Reply:
[Fixed](https://github.com/breez/breezmobile/pull/162/commits/96486619e86debc7d6514689a54adbb2e76a63bd)